pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+::  chat-cli: cli chat client using chat-store and friends
+::
+::    pulls all known messages into a single stream.
+::    type ;help for usage instructions.
+::
+::    note that while the chat-store only cares about paths,
+::    we mostly deal with [ship path] (aka target) here.
+::    when sending messages (through the chat hook),
+::    we concat the ship onto the head of the path,
+::    and trust it to take care of the rest.
+::
+/-  sole-sur=sole, *chat-store, *chat-view, *chat-hook
+/+  sole-lib=sole
+/=  seed  /~  !>(.)
+::
+|%
++$  state
+  $:  grams=(list mail)                             ::  all messages
+      known=(set [target serial])                   ::  known message lookup
+      count=@ud                                     ::  (lent grams)
+      bound=(map target char)                       ::  bound circle glyphs
+      binds=(jug char target)                       ::  circle glyph lookup
+      audience=(set target)                         ::  active targets
+      settings=(set term)                           ::  frontend flags
+      width=@ud                                     ::  display width
+      timez=(pair ? @ud)                            ::  timezone adjustment
+      cli=[=bone state=sole-share:sole-sur]         ::  console id & state
+  ==
+::
++$  mail  [source=target envelope]
++$  target  [=ship =path]
+::
++$  glyph  char
+++  glyphs  "!@#$%^&()-=_+[]\{};'\\:\",.<>?"
+::
++$  command
+  $%  [%target (set target)]                        ::  set messaging target
+      [%say (list letter)]                          ::  send message
+      [%eval cord hoon]                             ::  send #-message
+    ::
+      [%create path =(unit glyph)]                  ::  create chat
+      [%join target =(unit glyph)]                  ::  join target
+      [%leave target]                               ::  nuke target
```

Comment:
Can you add a [%delete target] that completely deletes state using %chat-view's %remove command,
and have [%leave target] use %chat-hook's %remove command?

Reply:
Huh, I thought `%chat-hook`'s `%remove` also did a complete delete of the local data, making `%leave` essentially equivalent to `%delete`.

Reply:
Nope!